src/crypto/sodium.cpp

```diff
@@ -107,52 +121,145 @@ int EdDSA::verify(const unsigned char* msg, unsigned long long msglen,
                                   signedmsg, msglen + crypto_sign_BYTES, pubKey);
     free(msgbuffer);
     free(signedmsg);
-    if (result) {
-        return(1);
-    } else {
-        // crypto_sign_open() returns -1 on failure.
-        return(0);
-    }
+
+    // crypto_sign_open() returns 0 on success.
+    return result ? 1 : 0;
 }
 
 
-// Generates a new Ed25519 private key seed. The key seed is stored in the object.
-int EdDSA::genKeySeed(unsigned char* privKey) {
-    // Make space for a new key seed (if not present).
-    if (!this->keySeed) {
-        this->keySeed = (unsigned char*)malloc(crypto_sign_SEEDBYTES);
-        if (this->keySeed == NULL) {
+// Derives the Ed25519 private and public keys from the stored private key seed.
+int EdDSA::genKeys()
+{
+    int check = 1;  // error code != 0
```

Comment:
Documentation of libsodium: 
```int	crypto_sign_seed_keypair(unsigned char *pk, unsigned	 char *sk, const unsigned char	 *seed);``` They could add different values we may need to filter.